modules/library/render/src/main/java/org/geotools/renderer/label/LabelCacheImpl.java

```diff
@@ -1265,6 +1272,43 @@ private boolean paintPointLabelInternal(LabelPainter painter, AffineTransform te
             return true;
         }
     }
+    
+    /**
+     * Splits the polygon into its component lines and labels those, one by one
+     */
```

Comment:
But it's exactly what it does, extract the lines from the polygon and label them one by one. How would you word it?